Firebase/Messaging/FIRMessagingRemoteNotificationsProxy.m

```diff
@@ -23,44 +23,36 @@
 #import "FIRMessagingLogger.h"
 #import "FIRMessagingUtilities.h"
 #import "FIRMessaging_Private.h"
+#import <GoogleUtilities/GULAppDelegateSwizzler.h>
 
-static const BOOL kDefaultAutoRegisterEnabledValue = YES;
 static void * UserNotificationObserverContext = &UserNotificationObserverContext;
 
 static NSString *kUserNotificationWillPresentSelectorString =
     @"userNotificationCenter:willPresentNotification:withCompletionHandler:";
 static NSString *kUserNotificationDidReceiveResponseSelectorString =
     @"userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:";
-static NSString *kReceiveDataMessageSelectorString = @"messaging:didReceiveMessage:";
 
-@interface FIRMessagingRemoteNotificationsProxy ()
+@interface FIRMessagingRemoteNotificationsProxy () <UIApplicationDelegate>
```

Comment:
Is there an alternative than following this protocol? This makes macOS support very hard .

Reply:
Refer comment below.

Reply:
@chliangGoogle Does FCM support MacOS now? If not, I can take care of MacOS support for `GULAppDelegateSwizzler` in a separate PR. What do you think?

Reply:
@maksymmalyhin FCM does not yet support MacOS. It recently added tvOS See https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseMessaging.podspec#L22

Reply:
@paulb777 It looks like tvOS has no difference in the App Delegate, so supporting tvOS by `GULAppDelegateSwizzler` should be trivial. I'll update it in this PR

Reply:
Will we make GULAppDelegateSwizzler macOS support in this PR? If not, we can add a todo and fix in another PR.

Reply:
I created a ticket for this #2706